src/OpenApi/JSON.hs

```diff
@@ -0,0 +1,87 @@
+module OpenApi.JSON
+  ( generateRenamed
+  , parseJSONFixed
+  , parseRefSum
+  , parseRenamed
+  )
+where
+
+import Data.List (lookup)
+import Data.String (String)
+import GHC.Enum (Bounded, Enum, minBound)
+import GHC.Generics (Generic, Rep)
+import OpenApi.Prelude
+
+import qualified Data.Aeson.Types    as JSON
+import qualified Data.HashMap.Strict as HashMap
+import qualified Data.Text           as Text
+
+parseRenamed
+  :: (Generic a, JSON.GFromJSON JSON.Zero (Rep a))
+  => [(String, String)]
+  -> JSON.Value
+  -> JSON.Parser a
+parseRenamed renames = JSON.genericParseJSON $ renameOptions renames
+
+generateRenamed
+  :: (Generic a, JSON.GToJSON JSON.Zero (Rep a))
+  => [(String, String)]
+  -> a
+  -> JSON.Value
+generateRenamed renames = JSON.genericToJSON $ renameOptions renames
+
+parseJSONFixed
+  :: forall a b . (Bounded b, Enum b, Eq a, Show a)
+  => Text
+  -> (String -> (a -> JSON.Parser b) -> JSON.Value -> JSON.Parser b)
```

Comment:
You could add a `c` parameter and replace `JSON.Value` with `c` here.. as well as on line 38.

Reply:
good call, fixed.